src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -455,11 +455,10 @@ public function getArrayType(Expr\Array_ $expr, callable $getTypeCallback): Type
 				} else {
 					$arrayBuilder->degradeToGeneralArray();
 
-					if (! (new StringType())->isSuperTypeOf($valueType->getIterableKeyType())->no() && $this->phpVersion->supportsArrayUnpackingWithStringKeys()) {
-						$arrayBuilder->setOffsetValueType($valueType->getIterableKeyType(), $valueType->getIterableValueType());
-					} else {
-						$arrayBuilder->setOffsetValueType(new IntegerType(), $valueType->getIterableValueType(), !$valueType->isIterableAtLeastOnce()->yes() && !$valueType->getIterableValueType()->isIterableAtLeastOnce()->yes());
-					}
+					$offsetType = !(new StringType())->isSuperTypeOf($valueType->getIterableKeyType())->no() && $this->phpVersion->supportsArrayUnpackingWithStringKeys()
```

Comment:
nit: maybe the obvious faster condition, namely the `supportsArrayUnpackingWithStringKeys` should be done first

Reply:
sure, that makes sense, thx